profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java

```diff
@@ -71,6 +74,10 @@ public void export(RecordedEvent event, AllocationEventSampler sampler, SpanCont
     if (sampler != null) {
       sampler.addAttributes((k, v) -> builder.put(k, v), (k, v) -> builder.put(k, v));
     }
+    if (stackTrace.isTruncated() || stackTrace.getFrames().size() > stackDepth) {
+      builder.put(THREAD_STACK_TRUNCATED, true);
```

Comment:
Maybe add a comment explaining that the actual truncation happens in `StackTraceParser`? For a bit I was confused that it's missing.

Reply:
usually truncation is already done by jfr, truncation in `StackTraceParser` is only used when jfr stack depth is greater than ours

Reply:
Right, I understand that. What I mean is that it was very easy to find this logic for "hey, if JFR gives us a stack trace taller than we want, truncate it" in case of pprof allocations, but for other type/format it is less obvious.

Reply:
This thread can be resolved